components/uniset/src/uniset.rs

```diff
@@ -124,25 +125,52 @@ impl UnicodeSet {
         &self.inv_list
     }
 
-    /// Yields an iterator going through the character set in the [`UnicodeSet`]
+    /// Yields an [`Iterator`] going through the character set in the [`UnicodeSet`]
     ///
     /// # Examples
     ///
     /// ```
     /// use icu::uniset::UnicodeSet;
     /// let example_list = vec![65, 68, 69, 70];
     /// let example = UnicodeSet::from_inversion_list(example_list).unwrap();
-    /// let mut example_iter = example.iter_chars();
-    /// assert_eq!(Some('A'), example_iter.next());
-    /// assert_eq!(Some('B'), example_iter.next());
-    /// assert_eq!(Some('C'), example_iter.next());
-    /// assert_eq!(Some('E'), example_iter.next());
-    /// assert_eq!(None, example_iter.next());
+    /// let mut ex_iter_chars = example.iter_chars();
+    /// assert_eq!(Some('A'), ex_iter_chars.next());
+    /// assert_eq!(Some('B'), ex_iter_chars.next());
+    /// assert_eq!(Some('C'), ex_iter_chars.next());
+    /// assert_eq!(Some('E'), ex_iter_chars.next());
+    /// assert_eq!(None, ex_iter_chars.next());
     /// ```
     pub fn iter_chars(&self) -> impl Iterator<Item = char> + '_ {
         self.inv_list.chunks(2).flat_map(|pair| (pair[0]..pair[1])).filter_map(char::from_u32)
     }
 
+    /// Yields an [`Iterator`] returning the ranges of the code points that are
+    /// included in the [`UnicodeSet`]
+    ///
+    /// Ranges are returned as [`RangeInclusive`], which is inclusive of its
+    /// `end` bound value. An end-inclusive behavior matches the ICU4C/J
+    /// behavior of ranges, ex: `UnicodeSet::contains(UChar32 start, UChar32 end)`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use icu::uniset::UnicodeSet;
+    /// let example_list = vec![65, 68, 69, 70];
+    /// let example = UnicodeSet::from_inversion_list(example_list).unwrap();
+    /// let mut example_iter_ranges = example.iter_ranges();
+    /// assert_eq!(Some(65..=67), example_iter_ranges.next());
+    /// assert_eq!(Some(69..=69), example_iter_ranges.next());
+    /// assert_eq!(None, example_iter_ranges.next());
+    /// ```
+    pub fn iter_ranges(&self) -> impl Iterator<Item = RangeInclusive<u32>> + '_ {
```

Comment:
suggestion (nb): probably could make this return `impl ExactSizeIterator`, unsure if it'll actually work

Reply:
Done. Yep, that worked.

(I found out for my personal knowledge that it worked because the struct returned by the `.map()` operation on an Iterator (`std::iter::Map`) implements ExactSizeIterator. And by the same reasoning, trying to make a similar change on a similar method from `pub fn iter_chars(&self) -> impl Iterator<...>` to have the return type `... -> impl ExactSizeIterator` doesn't work because the code in the method body returns a call of `.filter_map()` on the Iterator, which returns a `std::iter::FilterMap`, which doesn't implement ExactSizeIterator.)